azure-iot-device/azure/iot/device/iothub/models/message.py

```diff
@@ -52,3 +53,6 @@ def __init__(
         self.content_encoding = content_encoding
         self.content_type = content_type
         self.output_name = output_name
+
+    def __str__(self):
+        return str(self.data)
```

Comment:
>         [](start = 0, length = 8)

For MQTT data can be string , number or bytes. But in other cases if it can be a complex class this will only work if that class also has a dunder string method defined right ? I see in python it is not a practice to define dunder string , will this pose a problem ?

Reply:
In the absence of `__str__` it defaults to `__repr__`, which comes automatically defined for every type of object.

---
In reply to: [290987260](https://github.com/Azure/azure-iot-sdk-python-preview/pull/101#discussion_r290987260) [](ancestors = 290987260)